src/coreComponents/linearAlgebra/interfaces/hypre/HypreSuperlu.cpp

```diff
@@ -0,0 +1,286 @@
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file HypreSuperlu.cpp
+ */
+
+#include "HypreSuperlu.hpp"
+#include "common/Stopwatch.hpp"
+
+#include "HYPRE.h"
+#include "_hypre_parcsr_mv.h"
+
+namespace geosx
+{
+
+// Check matching requirements on index/value types between GEOSX and SuperLU_Dist
+
+static_assert( sizeof( int_t ) == sizeof( globalIndex ),
+               "SuperLU_Dist int_t and geosx::globalIndex must have the same size" );
+
+static_assert( std::is_signed< int_t >::value == std::is_signed< globalIndex >::value,
+               "SuperLU_Dist int_t and geoex::globalIndex must both be signed or unsigned" );
+
+static_assert( std::is_same< double, real64 >::value,
+               "SuperLU_Dist real and geosx::real64 must be the same type" );
+
+void ConvertToSuperMatrix( HypreMatrix const & matrix,
+                           SuperLU_DistData & SLUDData )
+{
+  // Merge diag and offd into one matrix (global ids)
+  SLUDData.localStrip = hypre_MergeDiagAndOffd( matrix.unwrapped() );
+
+  HYPRE_Int const * const hypreI = hypre_CSRMatrixI( SLUDData.localStrip );
+  SLUDData.rowPtr = new int_t[matrix.numLocalRows()+1];
+  for( localIndex i = 0; i <= matrix.numLocalRows(); ++i )
+  {
+    SLUDData.rowPtr[i] = LvArray::integerConversion< int_t >( hypreI[i] );
+  }
+
+  dCreate_CompRowLoc_Matrix_dist( &SLUDData.mat,
+                                  toSuperlu_intT( matrix.numGlobalRows() ),
+                                  toSuperlu_intT( matrix.numGlobalRows() ),
+                                  toSuperlu_intT( matrix.numLocalNonzeros() ),
+                                  toSuperlu_intT( matrix.numLocalRows() ),
+                                  toSuperlu_intT( matrix.ilower() ),
+                                  hypre_CSRMatrixData( SLUDData.localStrip ),
+                                  toSuperlu_intT( hypre_CSRMatrixBigJ( SLUDData.localStrip ) ),
+                                  SLUDData.rowPtr,
+                                  SLU_NR_loc,
+                                  SLU_D,
+                                  SLU_GE );
+}
+
+void SuperLU_DistCreate( HypreMatrix const & matrix,
+                         LinearSolverParameters const & params,
+                         SuperLU_DistData & SLUDData )
+{
+  // Initialize options.
+  set_default_options_dist( &SLUDData.options );
+  if( params.logLevel > 1 )
+  {
+    SLUDData.options.PrintStat = YES;
+  }
+  else
+  {
+    SLUDData.options.PrintStat = NO;
+  }
```

Comment:
A matter of personal taste, but I find it more readable to rewrite all of these binary branches using ternary operator; up to you though.

Reply:
You're right. It's more readable.